flink-table/flink-sql-parser/src/main/java/org/apache/flink/sql/parser/ddl/SqlAlterModelRename.java

```diff
@@ -0,0 +1,67 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.flink.sql.parser.ddl;
+
+import org.apache.calcite.sql.SqlIdentifier;
+import org.apache.calcite.sql.SqlNode;
+import org.apache.calcite.sql.SqlWriter;
+import org.apache.calcite.sql.parser.SqlParserPos;
+import org.apache.calcite.util.ImmutableNullableList;
+
+import java.util.List;
+
+import static java.util.Objects.requireNonNull;
+
+/** ALTER MODEL [IF EXISTS] [[catalogName.] dataBasesName.]modelName RENAME TO newModelName. */
+public class SqlAlterModelRename extends SqlAlterModel {
+
+    private final SqlIdentifier newModelName;
+
+    public SqlAlterModelRename(
+            SqlParserPos pos,
+            SqlIdentifier modelName,
+            SqlIdentifier newModelName,
+            boolean ifModelExists) {
+        super(pos, modelName, ifModelExists);
+        this.newModelName = requireNonNull(newModelName, "newModelName should not be null");
+    }
+
+    public SqlIdentifier getNewModelName() {
+        return newModelName;
+    }
+
+    public String[] fullNewModelName() {
+        if (newModelName != null) {
+            return newModelName.names.toArray(new String[0]);
+        }
+        return new String[0];
+    }
+
+    @Override
+    public List<SqlNode> getOperandList() {
+        return ImmutableNullableList.of(modelName, newModelName);
```

Comment:
nit:
```suggestion
        return List.of(modelName, newModelName);
```
it's better to avoid using guava if possible.
Since we moved to java 11, now can use `List.of`